frontend/src/components/pages/GlobalForms/WaiverTab/WaiverSectionCard.tsx

```diff
@@ -1,50 +1,180 @@
 import React from "react";
-import { Text, Box, Flex, HStack, Spacer } from "@chakra-ui/react";
+import {
+  Text,
+  Box,
+  Flex,
+  HStack,
+  Spacer,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  Textarea,
+  Checkbox,
+} from "@chakra-ui/react";
 import { WaiverClause } from "../../../../types/AdminTypes";
 
 interface WaiverSectionCardProps {
   clauseIdx: number;
   clauseData: WaiverClause;
+  onEditWaiverSection: (
+    clauseText: string,
```

Comment:
Instead of listing the fields in the `onEditWaiverSection` function here, could you instead pass in a `WaiverClause` object? The benefit of this is that if we change the definition of the WaiverClause, then we don't need to update the function everywhere with the new fields etc.